### PR TITLE
Added Cartel Faction

### PR DESCRIPTION
**Pull request by TyroneMF**
Completed UI previews
Seperated sub-categories into seperate HPP files.

### CODE REVIEW

File: addons/cartel/CfgGroups.hpp [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {10,-10,0};
+                    };
+                };
+
+                class TACU_Cartel_G_Soldado_Squad {
+                    name = "Soldados Squad";
+                    side = 0;
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\b_inf.paa";
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                    icon = "\A3\ui_f\data\map\markers\nato\o_inf.paa";
```

File: addons/cartel/CfgGroups.hpp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                name = "Soldados";
+
+                class TACU_Cartel_G_Soldado_Patrol {
+                    name = "Patrol";
+                    side = 0;
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\b_inf.paa";
+                    class unit0 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_01";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "SERGEANT";
+                        position[] = {0,0,0};
+                    };
+                    class unit1 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_02";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {5,-5,0};
+                    };
+                };
+
+                class TACU_Cartel_G_Soldado_Fireteam {
+                    name = "Soldado Fireteam";
+                    side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                    side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\b_inf.paa";
+                    class unit0 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_01";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "SERGEANT";
+                        position[] = {0,0,0};
+                    };
+                    class unit1 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_04";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {5,-5,0};
+                    };
+                    class unit2 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_AT";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {-5,-5,0};
+                    };
+                    class unit3 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_MG";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+                };
+
+                class TACU_Cartel_G_Soldado_Squad {
+                    name = "Soldados Squad";
+                    side = 0;
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\b_inf.paa";
+                    class unit0 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_01";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "SERGEANT";
+                        position[] = {0,0,0};
+                    };
+                    class unit1 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_02";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {5,-5,0};
+                    };
+                    class unit2 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_03";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {-5,-5,0};
+                    };
+                    class unit3 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_04";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {10,-10,0};
+                    };
+                    class unit4 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_AT";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgGroups.hpp [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+                        rank = "PRIVATE";
+                        position[] = {-10,-10,0};
+                    };
+                    class unit5 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_MG";
+                        side = 1;
```
Comment (Kresky) [inline addons/cartel/CfgGroups.hpp]:
```suggestion
                        side = 0;
```

File: addons/cartel/CfgVehicles.hpp [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+    // Base Classes
+    class TACU_Main_U_OPFOR_Soldier_Base;
+    class I_C_Offroad_02_unarmed_F;
+    class I_C_Offroad_02_LMG_F;
+    class C_Van_02_transport_F;
+
+    #include "CfgVehicles_Sicarios.hpp"
+    #include "CfgVehicles_Soldados.hpp"
+
+    // Vehicles - Cartel
```
Comment (Kresky) [inline addons/cartel/CfgVehicles.hpp]: Soldado vehicles, so can be put under `CfgVehicles_Soldados.hpp`

File: addons/cartel/CfgVehicles_Sicarios.hpp [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+        role = "Rifleman";
+        uniformClass = "tacs_Suit_VIP";
+        linkedItems[] = {DEFAULT_ITEMS_RADIO, "V_Rangemaster_belt", "G_Aviator"};
+        respawnLinkedItems[] = {DEFAULT_ITEMS_RADIO, "V_Rangemaster_belt", "G_Aviator"};
+        Items[] = {mag_5("ACE_fieldDressing")};
+        respawnItems[] = {mag_5("ACE_fieldDressing")};
+        weapons[] = {"CUP_hgun_Glock17", "Throw", "Put"};
+        respawnWeapons[] = {"CUP_hgun_Glock17", "Throw", "Put"};
+        magazines[] = {mag_4("CUP_17Rnd_9x19_glock17"), mag_2("HandGrenade"), "SmokeShell"};
+        respawnMagazines[] = {mag_4("CUP_17Rnd_9x19_glock17"), mag_2("HandGrenade"), "SmokeShell"};
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Sicarios.hpp]:
```suggestion
        respawnMagazines[] = {mag_4("CUP_17Rnd_9x19_glock17"), mag_2("HandGrenade"), "SmokeShell"};
        headgearList[] = {
            "",1
        };
```

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 17]
```diff
@@ -0,0 +1,131 @@
+    // Units - Cartel Soldados
+
+    class TACU_Cartel_U_Soldier_Base: TACU_Main_U_OPFOR_Soldier_Base {
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Soldados.hpp]: `TACU_Main_U_OPFOR_Soldier_Base` already acts as a base, so `TACU_Cartel_U_Soldier_Base` can be replaced with `TACU_Cartel_U_Soldado_Rifleman_01` (and following Cartel Soldados can be based on rifleman 01)

**Comment by TyroneMF** [comment]
All changes implemented.

File: addons/cartel/CfgGroups.hpp [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+                class TACU_Cartel_G_Soldado_Patrol {
+                    name = "Patrol";
+                    side = 0;
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\o_inf.paa";
+                    class unit0 {
+                        vehicle = "TACU_Cartel_U_Soldado_Rifleman_01";
+                        side = 0;
+                        rank = "SERGEANT";
+                        position[] = {0,0,0};
```
Comment (jonpas) [inline addons/cartel/CfgGroups.hpp]: Spaces after commas.

File: addons/cartel/CfgGroups.hpp [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+        class TACU_Cartel {
+            name = "Cartel";
+            class TACU_Cartel_G_Soldado_Infantry {
+                name = "Soldados";
+
+                class TACU_Cartel_G_Soldado_Patrol {
+                    name = "Patrol";
+                    side = 0;
+                    faction = "TACU_Cartel";
+                    icon = "\A3\ui_f\data\map\markers\nato\o_inf.paa";
```
Comment (jonpas) [inline addons/cartel/CfgGroups.hpp]: Prefer lower-case `\a3...`.

File: addons/cartel/CfgFactionClasses.hpp [hunk 20]
```diff
@@ -0,0 +1,16 @@
+class CfgFactionClasses {
+    class TACU_Cartel {
+        displayName = "Cartel";
```
Comment (jonpas) [inline addons/cartel/CfgFactionClasses.hpp]: Are you sure we want it just as "Cartel"? That's very general, is nothing else already using that name?

Comment (TyroneMF) [reply addons/cartel/CfgFactionClasses.hpp]: Correct, No other cartel factions currently exist.

File: addons/cartel/stringtable.xml [hunk 21]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project name="TACU">
+    <Package name="Cartel">
+        <Key ID="STR_TACU_Cartel_Author">
+            <English>Tyrone</English>
```
[the remark this answers is not in the document]
Comment (TyroneMF) [reply addons/cartel/stringtable.xml]: Same as all other Stringtables

File: addons/cartel/CfgFactionClasses.hpp (hunk 20 above)
Comment (jonpas) [reply addons/cartel/CfgFactionClasses.hpp]: Quickly reserve it then! :laughing:

**Comment by TyroneMF** [comment]
All changes implemented.

File: addons/cartel/stringtable.xml (hunk 21 above)
[the remark this answers is not in the document]
Comment (Kresky) [reply addons/cartel/stringtable.xml]: Hm, I suppose we can look at removing the stringtable overall, given we're not really adding new items for the most part.

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+        Items[] = {mag_5("ACE_fieldDressing")};
+        respawnItems[] = {mag_5("ACE_fieldDressing")};
+        weapons[] = {"hgun_P07_F", "Throw", "Put"};
+        respawnWeapons[] = {"hgun_P07_F", "Throw", "Put"};
+        magazines[] = {mag_7("16Rnd_9x21_Mag"), mag_2("HandGrenade"), "SmokeShell"};
+        respawnMagazines[] = {mag_7("16Rnd_9x21_Mag"), mag_2("HandGrenade"), "SmokeShell"};
+        editorSubcategory = "TACU_Cartel_EdSubCat_Soldados";
+        editorPreview = QPATHTOF(ui\Cartel_U_Soldado_Rifleman.jpg);
+        headgearList[] = {
+        "H_Bandanna_gry", 0.30,
```
Comment (jonpas) [inline addons/cartel/CfgVehicles_Soldados.hpp]: Indent!

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 23]
```diff
@@ -25,10 +25,10 @@
...
         editorPreview = QPATHTOF(ui\Cartel_U_Soldado_Rifleman.jpg);
         headgearList[] = {
-        "H_Bandanna_gry", 0.30,
-        "H_Bandanna_cbr", 0.30,
-        "H_Bandanna_khk", 0.30
-        };
+            "H_Bandanna_gry", 0.30,
+            "H_Bandanna_cbr", 0.30,
+            "H_Bandanna_khk", 0.30
+		};
```
Comment (jonpas) [inline addons/cartel/CfgVehicles_Soldados.hpp]: Where did you pull tabs from? :P

Comment (TyroneMF) [reply addons/cartel/CfgVehicles_Soldados.hpp]: I actually have no idea. o.O

File: addons/cartel/config.cpp [hunk 24]
```diff
@@ -0,0 +1,39 @@
...
+            "TACU_Cartel_U_Sicario_Rifleman_MG",
+            "TACU_Cartel_U_Sicario_Grenadier",
+            "TACU_Cartel_V_Offroad_Jeep",
+            "TACU_Cartel_V_Offroad_Jeep_Armed",
+            "TACU_Cartel_V_Van_Transport"
+        };
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"tacu_main"};
+        author = ECSTRING(Cartel,Author);
```
Comment (Kresky) [inline addons/cartel/config.cpp]:
```suggestion
        author = "Tyrone";
```

File: addons/cartel/config.cpp [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+        authors[] = {ECSTRING(Cartel,Author)};
```
Comment (Kresky) [inline addons/cartel/config.cpp]:
```suggestion
        authors[] = {"Tyrone"};
```

File: addons/cartel/CfgVehicles_Sicarios.hpp [hunk 26]
```diff
@@ -0,0 +1,99 @@
+    // Units - Cartel Sicarios
+
+    class TACU_Cartel_U_Sicario_Rifleman_01: TACU_Main_U_OPFOR_Soldier_Base {
+        author = ECSTRING(Cartel,Author);
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Sicarios.hpp]:
```suggestion
        author = "Tyrone";
```

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 27]
```diff
@@ -0,0 +1,157 @@
+    // Units - Cartel Soldados
+
+    class TACU_Cartel_U_Soldado_Rifleman: TACU_Main_U_OPFOR_Soldier_Base {
+        author = ECSTRING(Cartel,Author);
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Soldados.hpp]:
```suggestion
        author = "Tyrone";
```

File: addons/cartel/CfgVehicles_Sicarios.hpp [hunk 28]
```diff
@@ -0,0 +1,99 @@
...
+        respawnMagazines[] = {mag_2("CUP_30Rnd_Sa58_M_TracerG"), mag_2("HandGrenade"), "SmokeShell"};
+        editorPreview = QPATHTOF(ui\Cartel_U_Sicario_Rifleman_04.jpg);
+    };
+
+    class TACU_Cartel_U_Sicario_Rifleman_AT: TACU_Cartel_U_Sicario_Rifleman_01 {
+        displayName = "Sicario (AT)";
+        uniformClass = "tacs_Uniform_Combat_LS_BS_GP_TB";
+        linkedItems[] = {DEFAULT_ITEMS_RADIO, "V_PlateCarrierIA1_dgtl", "G_Aviator"};
+        respawnLinkedItems[] = {DEFAULT_ITEMS_RADIO, "V_PlateCarrierIA1_dgtl", "G_Aviator"};
+        weapons[] = {"CUP_arifle_Sa58RIS2", "Launch_RPG7_F", "Throw", "Put"};
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Sicarios.hpp]:
```suggestion
        weapons[] = {"CUP_arifle_Sa58RIS2", "CUP_launch_RPG7V", "Throw", "Put"};
```

File: addons/cartel/CfgVehicles_Sicarios.hpp [hunk 29]
```diff
@@ -0,0 +1,99 @@
...
+        respawnWeapons[] = {"CUP_arifle_Sa58RIS2", "Launch_RPG7_F", "Throw", "Put"};
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Sicarios.hpp]:
```suggestion
        respawnWeapons[] = {"CUP_arifle_Sa58RIS2", "CUP_launch_RPG7V", "Throw", "Put"};
```

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 30]
```diff
@@ -0,0 +1,157 @@
...
+        respawnMagazines[] = {mag_7("30Rnd_545x39_Mag_F"), mag_2("HandGrenade"), "SmokeShell"};
+        editorPreview = QPATHTOF(ui\Cartel_U_Soldado_Driver.jpg);
+    };
+
+    class TACU_Cartel_U_Soldado_Rifleman_AT: TACU_Cartel_U_Soldado_Rifleman_01 {
+        displayName = "Soldado (AT)";
+        uniformClass = "tacs_Uniform_Polo_TP_BS_TP_TB_NoLogo";
+        linkedItems[] = {DEFAULT_ITEMS_RADIO, "V_BandollierB_blk"};
+        respawnLinkedItems[] = {DEFAULT_ITEMS_RADIO, "V_BandollierB_blk"};
+        weapons[] = {"arifle_AKS_F", "Launch_RPG7_F", "Throw", "Put"};
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Soldados.hpp]:
```suggestion
        weapons[] = {"arifle_AKS_F", "CUP_launch_RPG7V", "Throw", "Put"};
```

File: addons/cartel/CfgVehicles_Soldados.hpp [hunk 31]
```diff
@@ -0,0 +1,157 @@
...
+        respawnWeapons[] = {"arifle_AKS_F", "Launch_RPG7_F", "Throw", "Put"};
```
Comment (Kresky) [inline addons/cartel/CfgVehicles_Soldados.hpp]:
```suggestion
        respawnWeapons[] = {"arifle_AKS_F", "CUP_launch_RPG7V", "Throw", "Put"};
```

File: addons/cartel/config.cpp [hunk 32]
```diff
@@ -28,8 +28,8 @@ class CfgPatches {
         weapons[] = {};
         requiredVersion = REQUIRED_VERSION;
         requiredAddons[] = {"tacu_main"};
-        author = ECSTRING(Cartel,Author);
-        authors[] = {ECSTRING(Cartel,Author)};
+        author = "Tyrone";
```
Comment (jonpas) [inline addons/cartel/config.cpp]: This one should be `ECSTRING(main,Author)`.

Comment (Kresky) [reply addons/cartel/config.cpp]: Ah, my bad. Both "author" and "authors"? (Will correct the other ones as well then)

Comment (jonpas) [reply addons/cartel/config.cpp]: No, just `author`. That way you have author of component in-game show up as "TAC Units Team", but the list of all authors show names of individuals.

Comment (Kresky) [reply addons/cartel/config.cpp]: Gotcha, will update the other ones as well to have ECSTRING.
@TyroneMF sorry about that, edit this one back as JJ mentioned, then everything should be good to be merged I believe.

Comment (jonpas) [reply addons/cartel/config.cpp]: It's just consistency, it's how we do it in CBA, ACE3, Theseus Services...

File: addons/cartel/config.cpp [hunk 33]
```diff
@@ -0,0 +1,39 @@
...
+            "TACU_Cartel_U_Sicario_Rifleman_MG",
+            "TACU_Cartel_U_Sicario_Grenadier",
+            "TACU_Cartel_V_Offroad_Jeep",
+            "TACU_Cartel_V_Offroad_Jeep_Armed",
+            "TACU_Cartel_V_Van_Transport"
+        };
+        weapons[] = {};
+        requiredVersion = REQUIRED_VERSION;
+        requiredAddons[] = {"tacu_main"};
+        author[] = "ECSTRING(Main,Author)";
```
Comment (Kresky) [inline addons/cartel/config.cpp]:
```suggestion
        author = ECSTRING(main,Author);
```

File: addons/cartel/config.cpp [hunk 34]
```diff
@@ -0,0 +1,39 @@
...
+        authors = "Tyrone";
```
Comment (Kresky) [inline addons/cartel/config.cpp]:
```suggestion
        authors[] = {"Tyrone"};
```

**Pull request merged**